lib/ci-helper.ts

```diff
@@ -554,6 +554,12 @@ export class CIHelper {
 
                 const userInfo = await this.getUserInfo(comment.author);
 
+                const commitProblems = await this.checkCommits(pr);
+                if (commitProblems) {
+                    this.github.addPRComment(pullRequestURL, commitProblems);
+                    throw new Error(`Max commits exceeded.`);
```

Comment:
Maybe the error should repeat the `commitProblems` rather than assuming that the problem was "too many commits"?

Reply:
This is the first pass in the linting and handling other errors was considered but not present.  Need to make sure we agree on reporting.  I was heading in the direction of one comment per commit which has all the suggestions for improving the commit.  This could cause multiple comments but they can be addressed individually.  One comment per suggestion seems excessive since the comments will show up each time a change is pushed.

So, the second pass in linting would pass a call back to checkCommits for use during the various tests.  I like the idea of passing an object to the call back with a short log message and long comment.

Reply:
Added the callback and changed `checkCommits` to return true if no errors.  No error is thrown for comment problems.  The log will include the PR comment data.